src/test/codegen/slice-as_chunks.rs

```diff
@@ -0,0 +1,30 @@
+// no-system-llvm
+// compile-flags: -O
+#![crate_type = "lib"]
+#![feature(slice_as_chunks)]
+
+// CHECK-LABEL: @chunks4
+#[no_mangle]
+pub fn chunks4(x: &[u8]) -> &[[u8; 4]] {
+    // CHECK-NEXT: start:
+    // CHECK-NEXT: lshr i64 %x.1, 2
+    // CHECK-NOT: shl
+    // CHECK-NOT: mul
+    // CHECK-NOT: udiv
+    // CHECK-NOT: urem
+    // CHECK: ret
+    x.as_chunks().0
+}
+
+// CHECK-LABEL: @chunks4_with_remainder
+#[no_mangle]
+pub fn chunks4_with_remainder(x: &[u8]) -> (&[[u8; 4]], &[u8]) {
+    // CHECK: and i64 %x.1, -4
+    // CHECK: and i64 %x.1, 3
+    // CHECK: lshr exact
```

Comment:
I wanted to ensure I didn't accidentally introduce extra instructions with the refactor, so codegen test.  It did change, but only from `lshr`/`and`/`and` to `and`/`and`/`lshr exact`, which is a thoroughly uninteresting difference.